sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -109,8 +113,14 @@
         this.tenantId = tenantId;
         this.clientId = clientId;
         this.options = options;
-        if (clientId == null) {
-            this.publicClientApplication = null;
+    }
+
+    private PublicClientApplication getPublicClientApplication(boolean sharedTokenCacheCredential) {
+        if (publicClientApplication != null) {
+            return publicClientApplication;
```

Comment:
This method is called from numerous methods on the IdentityClient some which pass `sharedTokenCacheCredential` as `true` and some as `false`. Whichever calls this method first will cache the publicClientApplication and then all subsequent calls will effectively ignore the `sharedTokenCacheCredential` parameter. I'm guessing this doesn't matter in practice because certain credentials only call certain methods, and each credential has it's own instance of `IdentityClient`, and `IdentityClient` is internal. However, this makes the functionality of this class and the classes that use it interdependent in a non-obvious way which could easily be overlooked if we try to make optimizations like sharing an instance of `IdentityClient`. I think we should consider refactoring this.

Reply:
Opened an issue #10675

Reply:
This method is only called once per IdentityClient and there's exactly one IdentityClient instance per credential. 
Yes it can have nicer designs, as something we can do for #10675